### PR TITLE
[WIP] Add Ubuntu Focal and Update/Streamline Debian installers.

### PR DESCRIPTION
**Pull request by MikeyMJCO**

### CODE REVIEW

**Comment by MikeyMJCO** [comment]
Tested this on a new vanilla install of `Focal` also tested on an upgraded existing BuildKit box. (Was on 16.04, then 18.04, now 20.04)

File: bin/civi-download-tools [hunk 1]
```diff
@@ -398,6 +398,35 @@ function do_system_disco() {
   set +e
 }
 
+###############################################################################
+
+function do_system_focal() {
+  set -e
+    PACKAGES="acl git wget unzip zip mysql-server mysql-client php7.3-cli php7.3-imap php7.3-ldap php7.3-curl php7.3-mysql php7.3-intl php7.3-gd php7.3-dev php7.3-bcmath php7.3-mbstring php7.3-soap php7.3-zip php7.3-xml apache2 libapache2-mod-php7.3 nodejs php-pear libmcrypt-dev libreadline-dev curl"
```
Comment (seamuslee001) [inline bin/civi-download-tools:405]: @MikeyMJCO is there a reason your installing libmcrypt-dev?

Comment (MikeyMJCO) [reply bin/civi-download-tools:405]: I forgot to remove it!

Comment (MikeyMJCO) [reply bin/civi-download-tools]: removed.

**Comment by MikeyMJCO** [comment]
This has a couple of deviations,

- I bumped the PHP version for Focal from 7.2 to 7.3 and went back and bumped Xenial to 7.1 from 7.0. 
- Switched to using Ondřej Surý PHP to make it easier to get patched versions outside of those that Ubuntu ship - which is currently 7.4.* on Focal - I'd expect needing to maintain this for future Ubuntu and Debian versions if our PHP version lags behind the "current stable" version (more so for Ubuntu than for Debian)

File: bin/civi-download-tools [hunk 2]
```diff
@@ -318,7 +318,7 @@ function get_system_installer() {
 ###############################################################################
 function do_system_xenial() {
   set -e
-    PACKAGES="acl git wget unzip zip mysql-server mysql-client php7.0-cli php7.0-imap php7.0-ldap php7.0-curl php7.0-mysql php7.0-intl php7.0-gd php7.0-mcrypt php7.0-bcmath php7.0-mbstring php7.0-soap php7.0-zip php7.0-xml apache2 libapache2-mod-php7.0 nodejs-legacy npm"
+    PACKAGES="acl git wget unzip zip mysql-server mysql-client php7.1-cli php7.1-imap php7.1-ldap php7.1-curl php7.1-mysql php7.1-intl php7.1-gd php7.1-mcrypt php7.1-bcmath php7.1-mbstring php7.1-soap php7.1-zip php7.1-xml apache2 libapache2-mod-php7.1 nodejs-legacy npm"
```
Comment (seamuslee001) [inline bin/civi-download-tools:321]: @MikeyMJCO won't you need to include the ppa repository here to allow for 7.1 install?

Comment (MikeyMJCO) [reply bin/civi-download-tools:321]: Yep!

**Comment by seamuslee001** [comment]
@MikeyMJCO this has worked for me? should we merge this now? also I was wondering if we should put a message pointing to the docs on MySQL8 differences given that Focal comes with MySQL 8 installed by default

**Comment by seamuslee001** [comment]
@MikeyMJCO any reason not to merge this in?

**Comment by MikeyMJCO** [comment]
There's a slim chance the Debian changes are broken - I hit some issues. Lemme retest this tomorrow and see where we are!

**Comment by seamuslee001** [comment]
@MikeyMJCO any update on this one?

**Comment by MikeyMJCO** [comment]
Closing this - it will either be resubmitted with changes as a new PR or form part of my monster-CI-introducing PR that may happen :-) Either way it serves no purpose!

**Pull request closed without merging**